### PR TITLE
improve:WebGL2CmdFuncCopyTexImagesToTexture add option to use gl.texImage2D

### PR DESCRIPTION
**Pull request by oahc09**
improve:WebGL2CmdFuncCopyTexImagesToTexture add option to use gl.texImage2D when copy video data to texture
because texImage2D(consume <1ms) is faster than texSubImage2D(consume 19ms) when copy 1080p video frame data.

texImage2D_0 11ms
texSubImage2D_19ms

**Refer:** https://github.com/pixijs/pixijs/pull/6088

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by bluesky013 — APPROVED** [review]

File: cocos/gfx/webgl2/webgl2-commands.ts [hunk 1]
```diff
@@ -2756,6 +2756,20 @@ export function WebGL2CmdFuncExecuteCmds (device: WebGL2Device, cmdPackage: WebG
     } // for
 }
 
+function isUseTexImage2D (texImages: Readonly<TexImageSource[]>, regions: Readonly<BufferTextureCopy[]>) {
+    if (texImages.length > 1 || regions.length > 1) return false;
+    const isVideoElement = texImages[0] instanceof HTMLVideoElement;
+    if (isVideoElement) {
+        const videoElement = texImages[0] as HTMLVideoElement;
+        const isSameSize = regions[0].texOffset.x === 0
+        && regions[0].texOffset.x === 0
```
Comment (bluesky013) [inline cocos/gfx/webgl2/webgl2-commands.ts]: texOffset.y

Comment (oahc09) [reply cocos/gfx/webgl2/webgl2-commands.ts]: > texOffset.y

ok

File: cocos/gfx/webgl2/webgl2-commands.ts [hunk 2]
```diff
@@ -2774,11 +2788,16 @@ export function WebGL2CmdFuncCopyTexImagesToTexture (
 
     switch (gpuTexture.glTarget) {
     case gl.TEXTURE_2D: {
-        for (let k = 0; k < regions.length; k++) {
-            const region = regions[k];
-            gl.texSubImage2D(gl.TEXTURE_2D, region.texSubres.mipLevel,
-                region.texOffset.x, region.texOffset.y,
-                gpuTexture.glFormat, gpuTexture.glType, texImages[n++]);
+        if (isUseTexImage2D(texImages, regions)) {
```
Comment (star-e) [inline cocos/gfx/webgl2/webgl2-commands.ts]: Shall we rename 'isUseTexImage2D' to 'toUseTexImage2D'?

Comment (oahc09) [reply cocos/gfx/webgl2/webgl2-commands.ts]: yes we can

**Review by star-e — APPROVED** [review]

**Pull request merged**